src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java

```diff
@@ -0,0 +1,105 @@
+package org.dependencytrack.event.kafka.exception;
+
+import alpine.Config;
+import alpine.common.logging.Logger;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler;
+import org.datanucleus.api.jdo.exceptions.ConnectionInUseException;
+import org.datanucleus.store.query.QueryInterruptedException;
+import org.dependencytrack.common.ConfigKey;
+
+import javax.jdo.JDOOptimisticVerificationException;
+import java.net.SocketTimeoutException;
+import java.sql.SQLTransientException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.REPLACE_THREAD;
+import static org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_CLIENT;
+
+public class KafkaStreamsUncaughtExceptionHandler implements StreamsUncaughtExceptionHandler {
+
+    private record ExceptionOccurrence(Instant occurredFirstAt, int count) {
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaStreamsUncaughtExceptionHandler.class);
+
+    private final Map<Class<? extends Throwable>, ExceptionOccurrence> transientExceptionOccurrences;
+    private final Duration transientExceptionThresholdInterval;
+    private final int transientExceptionThresholdCount;
+
+    public KafkaStreamsUncaughtExceptionHandler() {
+        this(
+                Duration.parse(Config.getInstance().getProperty(ConfigKey.KAFKA_STREAMS_TRANSIENT_PROCESSING_EXCEPTION_THRESHOLD_INTERVAL)),
+                Config.getInstance().getPropertyAsInt(ConfigKey.KAFKA_STREAMS_TRANSIENT_PROCESSING_EXCEPTION_THRESHOLD_COUNT)
+        );
+    }
+
+    KafkaStreamsUncaughtExceptionHandler(final Duration transientExceptionThresholdInterval,
+                                         final int transientExceptionThresholdCount) {
+        this.transientExceptionOccurrences = new ConcurrentHashMap<>();
+        this.transientExceptionThresholdInterval = transientExceptionThresholdInterval;
+        this.transientExceptionThresholdCount = transientExceptionThresholdCount;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public StreamThreadExceptionResponse handle(final Throwable exception) {
+        final Throwable rootCause = ExceptionUtils.getRootCause(exception);
+
+        if (rootCause instanceof TimeoutException
+                || rootCause instanceof ConnectTimeoutException
```

Comment:
Should we add ReadTimeoutException also ?

Reply:
Good point, will add it!

Reply:
Hmmm, there exists no `ReadTimeoutException`... But I do also recall seeing this class before, but perhaps in another code base.